flink-ml-parent/flink-ml-lib/src/main/java/org/apache/flink/ml/batchoperator/BatchOperator.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.flink.ml.batchoperator;
+
+import org.apache.flink.ml.api.misc.param.Params;
+import org.apache.flink.ml.batchoperator.source.TableSourceBatchOp;
+import org.apache.flink.ml.common.AlgoOperator;
+import org.apache.flink.table.api.Table;
+import org.apache.flink.util.Preconditions;
+
+/**
+ * Base class of batch algorithm operators.
+ *
+ * <p>This class is extended to support the data transmission between the BatchOperators.
+ */
+public abstract class BatchOperator<T extends BatchOperator<T>> extends AlgoOperator<T> {
```

Comment:
one suggestions I have is to put the operators into a single package
for example
```
|--operator
   |--AlgoOperator.java
   |--stream
      |--StreamOperator.java
      |--source
         |--TableSourceStreamOperator.java
   |--batch
      |--BatchOperator.java
      |--source
         |--TableSourceBatchOperator.java
```
IMO this is more clear than putting `AlgoOperator` in common. and others scatter around the package directory, what do you think?

Reply:
Thanks @becketqin , @walterddr , @ex00 , I have refactored the package paths according your suggestions.